### ISSUE TITLE
IE8: error thrown when opening and clicking in the overlay

### ISSUE THREAD

**Issue by roland-vachter** [Financial-Times/o-overlay#40]
Use case to reproduce: http://jsbin.com/nugifeceso/1/edit?html,output

The error:  'length' is null or not an object

**Comment by roland-vachter** [comment]
Is there a way I can see when will this be released as a new version? I can still see this issues.

**Comment by AlbertoElias** [comment]
Ah, the issue was closed automatically when merging in the pr. There will be a released done today

**Comment by roland-vachter** [comment]
Cool, thanks.

**Comment by Vallach** [comment]
Still seeing the issue on IE8 (also occurs when pressing the ESC key while the overlay is visible). Has the fixed version been released?

**Comment by AlbertoElias** [comment]
@Vallach Yes, it should be fixed. Can you pass me the information of how you're using it? Is it a real IE8 or a more modern IE in compatibility mode?

**Comment by Vallach** [comment]
I'm using Windows7 with native IE8 (8.0.7601.17514) and Windows XP, native IE8 (8.0.6001.18702), both on browser mode: IE8, document mode: IE8 Standards.

Issue occurs while using the "Commenting Settings" overlay available in the TEST env on any article page, above the comments widget (simply open the settings overlay and click anywhere inside it or press ESC).

'length' is null or not an object

**Comment by AlbertoElias** [comment]
@Vallach When you say this issue is still there, do you mean the module doesn't work as expected or that the you see the error in the console but the module works? The [commit](https://github.com/Financial-Times/o-overlay/commit/adfe84362bd74c49f737a1de880dad74e383be80) that fixes this mentions that the error is still thrown but the module should work as expected

**Comment by Vallach** [comment]
As described in my previous comment, the error is also thrown when pressing ESC. Not sure if just a coincidence, but it is not possible to close the overlay by pressing Escape. Works fine on other browsers where the "length" error is not encountered.

**Comment by AlbertoElias** [comment]
@Vallach I'm testing out the overlay demos in ie8 on Win7, and yes, when clicking Esc you get the error, but the overlay does close, it works as expected though you get the error, but you're expected to get the error https://github.com/Financial-Times/o-overlay/blob/master/README.md#browser-support

**Comment by roland-vachter** [comment]
Hi Alberto,

Please see this example:
http://jsbin.com/mebuxapixo/3
It reproduces the close by ESC button issue.

Also we noticed that through the modal background you still can click links and buttons, and you also can click the open button several times, which opens new overlays.

**Comment by AlbertoElias** [comment]
Hi Roland,

http://jsbin.com/raqaqu/1/edit This one does work, though you still get the error. I will try to see if there's a quick fix for the other option and for the modal issue, though, tbh, as ie8 is not primary anymore, I can't spend too much time on this

**Comment by AlbertoElias** [comment]
Oh, wait! I just noticed you were instantiating the module wrong. You need to do it like this:

```js
var dom = document.createElement('div');
         dom.innerHTML = 'test';
       var myOverlay = new bundles['o-overlay']('myoverlay', {html: dom, heading: {title: 'comments'}, modal: true});
       document.getElementById('open').addEventListener('click', function () {
         myOverlay.open();
       });
```

You're creating the same overlay over and over again every time you click the button and they clash. I will look into the modal problem

**Comment by roland-vachter** [comment]
Ok. As IE8 is not primary anymore, I think we shouldn't invest too much time in that, especially for such minor issues.

I will fix the instantiation issue.

**Comment by AlbertoElias** [comment]
Right, I'll add a note in the readme mentioning the modal background issue

**Comment by roland-vachter** [comment]
Currently I destroy the overlays when they are closed, and create a new one each time (but there's only 1 with that ID in the same time).
The issue is still there, so I think the issue is with the destroy (which seems to be just deleting the object, but it isn't removed from the array of objects).

**Comment by Vallach** [comment]
Noticed that 3 issues are discussed here:
1. 'length' is null or not an object console error
2. overlay does not close with ESC
3. links are clickable in the modal background

IE8 issues are low priority, still how are we going to handle these ones?

**Comment by AlbertoElias** [comment]
So issue 1 is unavoidable and doesn't make the overlay not work as expected
I'm talking with @roland-vachter as this seems to be an issue with your implementation to avoid this from happening
Regarding number 3, that's a separate issue and we decided as ie8 is low priority to not fix for now. I'll raise an issue for it and mark it as won't fix.

This issue is closed as there was an initial bug in o-overlay that was fixed. As number 2 is still an issue, I'm speaking privately with Roland as the conversation is smoother and it's regarding a specific implementation of the module. I'll comment the result afterwards.

Thanks a lot for the feedback
